### ISSUE TITLE
pass job id into worker function

### ISSUE THREAD

**Issue by derhuerst** [pgte/level-jobs#20]
thanks for this great lib!

i'm using `level-jobs` to expose a queue of compute-intensive tasks via a RESTful HTTP server. my tasks generate output files, so i need to put them in a per-job directory.

right now, i have to go to great lengths to have a random dir prefix:

- i generate random dir id
- i get the job id from `queue.push`
- i store the dir id under the job id in the leveldb database

this is a very complicated setup. getting the job id in the worker function would be perfect.

**Comment by pgte** [comment]
@derhuerst adding an argument to the worker function is not backwards-compatible, but, if you provide the PR, I'm willing to bump a major version of this that supports a worker function with the following signature:

```js
function work (id, payload, callback) {
  // do your work
}
```

I don't think that any of the following alternatives are good:

* adding a `.id` attribute to the work payload: we have no control over the type of the payload — it can be a number, an array, whatever. Adding an attribute to it may not work.,
* alternative of checking the worker function.length is a hack that can go very wrong because of some features of ES6.

**Comment by derhuerst** [comment]
> I don't think that any of the following alternatives are good:
>
> - adding a .id attribute to the work payload: we have no control over the type of the payload — it can be a number, an array, whatever. Adding an attribute to it may not work.,
> - alternative of checking the worker function.length is a hack that can go very wrong because of some features of ES6.

I agree, both are very hacky. Will prepare a PR.

**Comment by pgte** [comment]
Fixed by @derhuerst in https://github.com/pgte/level-jobs/pull/21, landed in v1.0.0.
